circuits/circuits/distribute/README.md

```diff
@@ -0,0 +1,48 @@
+# Circuits
+
+As well as being used to enforce the state transition function, ZKPs are used to handle distribution after shutdown.
+The distribution circuits take a set of claims over coins and output the rightful final owners given those claims.
+
+## Algorithm
+
+Distribution involves 3 parts:
+    - Filter out any claim that is invalid (i.e., did not occur in the state history)
+    - Resolve any conflicts between claims
+    - Prove that there are no conflicting claims in the final output
+
+For conceptual simplicity and good prover time, we will implement each part as a separate Nova circuit.
+
+### Filtering
+
+Claims are statments like `X owns coins [a,b] in block Y` along with a purported Merkle proof.
+The set of claims are represented as an accumulator, in particular a keccak hash chain.
+The state history is represented as a Merkle root, which is known onchain, and passed as a public input to the circuit. It must be passed unchanged through each iteration of the circuit.
+There is also an index public input, which is required to start at `0` and end at `claim_count`, which is also known onchain.
+There is a `filtered_count` input that starts at `0`.
+
+The circuit processes one claim per iteration/fold. The claim accumulator starts at 0, one claim is added per iteration, and it must end with the right onchain value.
```

Comment:
this "right onchain value" is `claim_count`?

Reply:
Ah, yes, I'll clarify that.